<?xml version="1.0"?>
<!-- AMBER 2D Model Description                                         -->
<robot name="amber3m">
  <material name="red">
    <color rgba="0 0 0.8 1"/>
  </material>


  <!-- Linkage Definitions -->
  <link name="Torso">
    <inertial>
      <origin xyz="0 0 -0.28507"/>
      <mass value="13"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0.1940" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.1 0.2 0.4999"/>
      </geometry>
      <origin xyz="0.0 0.0 -0.24995" rpy="0.0 0.0 0.0"/>
    </visual>
  </link>

  <link name="LeftThigh">
    <inertial>
      <origin xyz="0.0 0.0 -0.1494" rpy= "0 0 0"/>
      <mass value="3.4261"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0.0389" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <geometry>
        <cylinder length="0.4064" radius="0.02"/>
      </geometry>
      <origin xyz="0.0 -0.1 -0.2032" rpy="0.0 0.0 0.0"/>
    </visual>
  </link>
  <link name="LeftShin">
    <inertial>
      <origin xyz="0 0 -0.171885" rpy="0 0 0"/>
      <mass value="0.85216"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0.019286025" iyz="0" izz="0"/>
    </inertial>
    <collision name="LeftToe">
      <origin xyz="0.0 0 -0.4666" rpy="0 0 0"/>
      <geometry>
	    <sphere radius=".0001"/>
      </geometry>
    </collision>
    <visual>
      <geometry>
        <cylinder length="0.4667" radius="0.02"/>
      </geometry>
      <origin xyz="0.0 -0.1 -0.23335" rpy="0 0 0"/>
    </visual>
  </link>
  <link name="RightThigh">
    <inertial>
      <origin xyz="0.0 0.0 -0.1494" rpy= "0 0 0"/>
      <mass value="3.4261"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0.0389" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.1 -0.2032" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.4064" radius="0.02"/>
      </geometry>
    </visual>
  </link>
  <link name="RightShin">
    <inertial>
      <origin xyz="0.0 0.0 -0.171885" rpy="0 0 0"/>
      <mass value="0.85216"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0.019286025" iyz="0" izz="0"/>
    </inertial>
    <collision name="RightToe">
      <origin xyz="0.0 0 -0.4666" rpy="0 0 0"/>
      <geometry>
	    <sphere radius=".0001"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0.0 0.1 -0.23335" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.4667" radius="0.02"/>
      </geometry>
    </visual>
  </link>


  <!-- Joint Definitions -->


  <joint name="LeftHip" type="revolute">
    <parent link="Torso"/>
    <child link="LeftThigh"/>
    <origin xyz="0 0 -0.4999" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="400.0" lower="-1.6" upper="1.6" velocity="4"/>
  </joint>
  <joint name="LeftKnee" type="revolute">
    <parent link="LeftThigh"/>
    <child link="LeftShin"/>
    <origin xyz="0.0 0.0 -0.4064" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="400.0" lower="-0.1" upper="1.5" velocity="4"/>
  </joint>
    <joint name="RightHip" type="revolute">
    <parent link="Torso"/>
    <child link="RightThigh"/>
    <origin xyz="0.0 0.0 -0.4999" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="400.0" lower="-1.6" upper="1.6" velocity="4"/>
  </joint>
  <joint name="RightKnee" type="revolute">
    <parent link="RightThigh"/>
    <child link="RightShin"/>
    <origin xyz="0.0 0.0 -0.4064" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="400.0" lower="-0.1" upper="1.5" velocity="5"/>
  </joint>

  <!-- Robot Transmission Definitions -->

  <transmission name="LeftHipTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LeftHip">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="LeftHipMotor">
      <mechanicalReduction>91.4286</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>


  <transmission name="LeftKneeTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LeftKnee">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="LeftKneeMotor">
      <mechanicalReduction>91.4286</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <transmission name="RightHipTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RightHip">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="RightHipMotor">
      <mechanicalReduction>91.4286</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <transmission name="RightKneeTransmission">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RightKnee">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="RightKneeMotor">
      <mechanicalReduction>91.4286</mechanicalReduction>
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>

  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/amber3m</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>

  <gazebo reference="LeftShin">
    <mu1>.6</mu1>
    <mu2>.6</mu2>
    <material>Gazebo/Red</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>

  <gazebo reference="RightShin">
    <mu1>.6</mu1>
    <mu2>.6</mu2>
    <material>Gazebo/Red</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>

</robot>

